### PR TITLE
Feature trailers

### PR DESCRIPTION
**Pull request by rvanderwesthuizen**
- Added Youtube API to the app (model and repository)
- Used a WebKitView to embed a youtube video of the trailer of the selected movie/series

### CODE REVIEW

File: Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewModel/TrailerViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+//  TrailerViewModel.swift
+//  Mr Movie Info
+//
+//  Created by Ruan van der Westhuizen on 2021/11/24.
+//
+
+import Foundation
+
+class TrailerViewModel {
+    public weak var delegate: ViewModelDelegate?
```
Comment (bgouws) [inline Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewModel/TrailerViewModel.swift:11]: Think this can be private

**Review by bgouws — CHANGES_REQUESTED** [review]
Pretty clean 😄 just a little bit of clean up.

File: Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    
+    private var viewModel: TrailerViewModel!
+    
+    init(movieTitle: String) {
+        super.init(nibName: nil, bundle: nil)
+        viewModel = TrailerViewModel(movieTitle: movieTitle, repository: YoutubeRepository(), delegate: self)
+    }
+    
+    required init?(coder: NSCoder) {
```
[the remark this answers is not in the document]
Comment (rvanderwesthuizen) [reply Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift:23]: Yes, Xcode shouts at me if I don't

File: Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+class TrailerViewController: UIViewController {
+    @IBOutlet private weak var webView: WKWebView!
+    @IBOutlet private weak var returnButton: UIButton!
+    @IBOutlet private weak var activityIndicator: UIActivityIndicatorView!
+    
+    private var viewModel: TrailerViewModel!
+    
+    init(movieTitle: String) {
+        super.init(nibName: nil, bundle: nil)
+        viewModel = TrailerViewModel(movieTitle: movieTitle, repository: YoutubeRepository(), delegate: self)
```
[the remark this answers is not in the document]
Comment (rvanderwesthuizen) [reply Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift:20]: How do I make this a private lazy and still have it like this within the init?

File: Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (rvanderwesthuizen) [reply Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift:23]: Yes, Xcode gives me an error saying it needs this

File: Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (justinewright) [reply Mr Movie Info/Mr Movie Info/Feature-Trailers/ViewController/TrailerViewController.swift:20]: you need to move the movie title to a configure function, then you can make your view model lazy

**Review by bgouws — APPROVED** [review]

**Review by justinewright — APPROVED** [review]

**Pull request merged**